src/external/rawspeed/RawSpeed/Common.cpp

```diff
@@ -24,14 +24,20 @@
 
 
 #if defined(__APPLE__)
-#include <CoreServices/CoreServices.h>
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#include <stdlib.h>
+#include <string.h>
 
 int macosx_version()
 {
-  SInt32 gestalt_version;
   static int ver = 0; // cached
-  if (0 == ver && (Gestalt(gestaltSystemVersion, &gestalt_version) == noErr)) {
-    ver = gestalt_version;
+  char str[256];
+  size_t strsize = sizeof(str);
+  if (0 == ver && sysctlbyname("kern.osrelease", str, &strsize, NULL, 0) == 0) {
+    // sysctl is major.minor.path
+    *strchr(str, '.') = '\0';
```

Comment:
But what if the string does not contain `.`, e.g. if the string is just `\0`?
Please rewrite it to first find the `.` in a safe manner, checking for `strsize`, and then if it is there in the string bounds, replacing with `\0`.